modules/eventlog/src/main/java/com/intuit/wasabi/eventlog/impl/EventLogImpl.java

```diff
@@ -180,13 +196,27 @@ public void run() {
      * @param event the vent to prepare
      */
     /*test*/ void prepareEnvelope(final EventLogEvent event) {
-        if (event == null) {
-            return;
-        }
-        for (EventLogListener eventLogListener : listeners.keySet()) {
-            if (isSubscribed(eventLogListener, event)) {
-                eventPostThreadPoolExecutor.submit(new EventLogEventEnvelope(event, eventLogListener));
+        LOGGER.debug("preparing event: {}", event);
+
+        if (event != null) {
```

Comment:
I prefer fail fast with shallow nesting. But I guess it does not matter.
Also I do not know if you want to have this many debug log statements here, it feels more like a trace level.